### PR TITLE
Build for DDR on Win32 and 'display_no_mist' option

### PR DESCRIPTION
**Pull request by fassisrosa**

### CODE REVIEW

File: main/midnight/src/library/inc/mxtypes.h [hunk 1]
```diff
@@ -38,6 +38,9 @@ typedef std::function<void(void)> MXVoidCallback;
 #define RETURN_IF_NULL( x )  \
     if ( (x) == nullptr ) return
 
+#define RETURN_VALUE_IF_NULL(x, v ) \
```
Comment (fassisrosa) [inline main/midnight/src/library/inc/mxtypes.h:41]: This was the only change needed to build DDR on windows. Expecting a return value from the one method that compile complained about.

File: main/midnight/src/tme/scenarios/ddr/ddr_processor_text.cpp [hunk 2]
```diff
@@ -360,7 +360,7 @@ c_string tokens;
 
 std::string ddr_text::DescribeCharacterLocation( const mxcharacter* character )
 {
-    RETURN_IF_NULL(character);
+    RETURN_VALUE_IF_NULL(character, "");
```
Comment (fassisrosa) [inline main/midnight/src/tme/scenarios/ddr/ddr_processor_text.cpp:363]: Corresponding use of macro.

Comment (IcemarkUK) [reply main/midnight/src/tme/scenarios/ddr/ddr_processor_text.cpp:363]: I don't think this is needed. The original **should** have read RETURN_IF_NULL(character) "";

Comment (fassisrosa) [reply main/midnight/src/tme/scenarios/ddr/ddr_processor_text.cpp:363]: Makes sense. Making the change.

Comment (fassisrosa) [reply main/midnight/src/tme/scenarios/ddr/ddr_processor_text.cpp]: Done.

**Review by IcemarkUK — APPROVED** [review]

**Pull request merged**